### PR TITLE
chore(*): tweak priorities to remove eta_experiment in linear algebra

### PR DESCRIPTION
**Pull request by sgouezel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by sgouezel** [comment]
!bench

**Comment by kbuzzard** [comment]
This PR also solves the one remaining problem in #3805 (which looked kind of nasty)

File: Mathlib/test.lean
Comment (loefflerd) [inline Mathlib/test.lean:1]: Is this file supposed to be part of the PR, or was it perhaps included by mistake? The linter is unhappy with it

**Comment by kbuzzard** [comment]
Can someone explain the output of the bench command? I literally only want to check that green means "this PR has made the file compile faster" rather than "this PR has made the file compile slower"

**Comment by kbuzzard** [comment]
I took a look at FieldTheory.Subfield (one of the few files which was red). The declaration Subfield.inclusion is slower now: the DefEq trace change is basically this:
```
/-
 -- SG_test3
 [] [3.928642s] ✅ Subsemiring.toNonAssocSemiring T.toSubring.toSubsemiring =?= SubsemiringClass.toNonAssocSemiring T

 -- master
 [] [1.761194s] ✅ Subsemiring.toNonAssocSemiring T.toSubring.toSubsemiring =?= SubsemiringClass.toNonAssocSemiring T
-/
```
Is there some modification of the trick involving these Class typeclasses which can deal with this?

**Comment by kbuzzard** [comment]
This "change the order a bit and things get better" reminds me of this old thread https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Problem.20with.20type.20class.20search/near/161848744

**Comment by sgouezel** [comment]
!bench

**Comment by sgouezel** [comment]
Better version, updated to current mathlib4 and ready for review, in #3840

**Pull request closed without merging**